### ISSUE TITLE
Hessian calculation using CASSCF method

### ISSUE THREAD

**Issue by valievrashid** [nubakery/bagel#180]
Hello!
Previously we tried to use BAGEL for the CASPT2 transition state and had a problem. Now I tried to use the CASSCF hessian calculation and have a problem, too. The input file is attached. 
I can see the gradient calculation finished and then “the Segmentation fault” error appeared. I attached the h2o.out file, too. 
Previously we had a similar problem with the hessian calculation at CASPT2 level of theory. 
How can solve this problem?

**Comment by valievrashid** [comment]
I can't attach files. So this is input
 {"bagel" :[
 {
 "title" : "molecule",
 "basis" : "tzvpp",
 "angstrom": "true",
 "df_basis" : "tzvpp-jkfit",
 "geometry" : [
   { "atom" : "O", "xyz" : [     -0.507583898,      0.000000000,     -1.491839780 ] }, 
  { "atom" : "H", "xyz" : [     -0.507583898,      0.759337000,     -0.895796780 ] }, 
  { "atom" : "H", "xyz" : [     -0.507583898,     -0.759337000,     -0.895796780 ] }] 

},

{
  "title" : "hessian",
 
"nproc": 1,
"method" :[{
  "title" : "casscf",
  "nstate" : 1,
  "nact" : 6,
   "nclosed" : 2,
     "active" : [3, 4, 5, 6, 7, 8]

}]
},
{
  "title" : "print",
  "file" : "orbitals.molden",
  "orbitals" : true
},

{
  "title" : "print",
  "file" : "freq.molden",
  "vibration" : true
}

]}



-------------------------
This is output

  * process grid (1, 1) will be used
    master1.cyberia.tsu.ru          

  * using 24 threads per process

  ===============================================================
    BAGEL - Freshly leavened quantum chemistry                   
  ===============================================================

  *** Geometry ***

  { "atom" : "O", "xyz" : [     -0.959195,      0.000000,     -2.819169 ] },
  { "atom" : "H", "xyz" : [     -0.959195,      1.434939,     -1.692811 ] },
  { "atom" : "H", "xyz" : [     -0.959195,     -1.434939,     -1.692811 ] },

  Number of auxiliary basis functions:      113

  Since a DF basis is specified, we compute 2- and 3-index integrals:
    o Being stored without compression. Storage requirement is 0.003 GB
       - 3-index ints prep                         0.00
       - 3-index ints                              1.61
       - 2-index ints                              0.00
       - computing inverse                         0.13
        elapsed time:        1.75 sec.


  Number of basis functions:       59
  Number of electrons      :       10


    * METHOD: MOLECULE                             2.00

  The Hessian will be computed with central gradient differences (analytical gradients)
  Finite difference displacement (dx) is 0.00100000 bohr

**Comment by shiozaki** [comment]
@bessvlai Hi Bess, do you have any ideas as to what is going on? Hope you are well!

**Comment by shiozaki** [comment]
@jwpk1201 Hi Jae Woo, further inspection with Bess indicates this bug may have been introduced by your commit f5db2e6590c4bf607968ead1b0300cf3f452f36c. It could be wrong, but want to get your attention. Bess says she reproduced the error with 1-state CASSCF calc (please ask her if you cannot reproduce the problem).

https://github.com/nubakery/bagel/commit/f5db2e6590c4bf607968ead1b0300cf3f452f36c#diff-871ab7c2a74079e22a0c2209ab6a79e5

PS: I am not too comfortable with 
```
const int npass = natom * 3 / ncomm + 1;
```
This should be the following, isn't it?
```
const int npass = (natom * 3 - 1) / ncomm + 1;
```

**Comment by jwpk1201** [comment]
Hi Toru, Bess, and Rashid,

Hope you are well! I will take a look while I am traveling (until the end of the next week).

Sincerely,
Jae Woo

**Comment by shiozaki** [comment]
Hi Jae Woo,

There is another bug report on GitHub #183 that is to do with your edits on embarrassingly parallel mode. Could you take a look?

Toru

**Comment by jwpk1201** [comment]
Hi Toru,

OK, I will take a look. Probably line 53 should be placed after if (ncolor != 0) statements for that.

Jae Woo

**Comment by shiozaki** [comment]
More reasonable fix is to change that to
```
const int npass = (natom * 3 - 1) / ncomm + 1;
```
kind of things - in that way you will never have ncolor = 0. This happens in many places including this. You have to subtract 1 inside the bracket and add 1 afterwords.

**Comment by jwpk1201** [comment]
Hi Toru,

OK, I will fix all of those, and then test using Bess' input.

Jae Woo

**Comment by shiozaki** [comment]
This has been fixed by 05ae88b108a372b5ef7e7290dc0e03f528c05367
